typing/typemod.ml

```diff
@@ -889,11 +969,22 @@ and transl_signature env sg =
             mksig (Tsig_modtype mtd) env loc :: trem,
             sg :: rem,
             final_env
-        | Psig_open sod ->
-            let (_path, newenv, od) = type_open env sod in
-            let (trem, rem, final_env) = transl_sig newenv srem in
-            mksig (Tsig_open od) env loc :: trem,
-            rem, final_env
+        | Psig_open sod -> begin
+            let id, od = type_open env sod in
+            let open_env = od.open_env in
+            let (trem, rem, final_env) = transl_sig open_env srem in
+            match id with
+            | None -> mksig (Tsig_open od) env loc :: trem, rem, final_env
```

Comment:
I'd rewrite the code into something like:

````ocaml
let rem' = 
   match id with
   | None -> rem
   | Some ... -> ...
   ...
in
mksig (Tsig_open od) env loc :: trem, rem, final_env
````

Reply:
done